src/app/shared/table/table.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { TalentService } from '@talent/talent.service';
+import { AbstractListComponent } from '@shared/component/abstract-list.component';
+
+import { Talent } from '@shared/model/talent.model';
+import { SortOptions } from '@shared/service/model/sort-options.model';
+
+@Component({
+	selector: 'table',
+	templateUrl: './table.component.html',
+	styleUrls: ['./table.component.scss']
+})
+export class TableComponent {
+    private _sortBy: string;
+	private _sortOrder: number;
+
+	@Input() listOfItems: any;
+	@Output() sortOptions = new EventEmitter();
+	@Output() navigateToNew = new EventEmitter();
```

Comment:
naming for the events, they should define an action to easy understand what they mean, for example:

onSortChange, onNavigate